operator/controllers/service_controller.go

```diff
@@ -68,87 +71,105 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
+		serviceRun, err := r.createServiceRun(ctx, service)
 		if err != nil {
 			logger.Error(err, "Could not create ServiceRun")
 			return reconcile.Result{}, err
 		}
 
-		service.Status.Phase = v1alpha1.ServiceRunRunning
-		service.Status.ServiceRunName = serviceRunName
-
-		if err := r.Status().Update(ctx, service); err != nil {
-			logger.Error(err, "Could not update Service")
+		k8sEvent := r.generateK8sEvent(service, serviceRun)
+		if err := r.Create(ctx, k8sEvent); err != nil {
+			logger.Error(err, "Could not send serviceRun created K8s event")
 			return reconcile.Result{}, err
 		}
-		return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
-	}
 
-	logger.Info("Checking status")
+		return ctrl.Result{}, nil
+	}
 
-	serviceRun := &v1alpha1.ServiceRun{}
-	err = r.Get(ctx, types.NamespacedName{Name: service.Status.ServiceRunName, Namespace: service.Namespace}, serviceRun)
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("could not fetch ServiceRun: %+v", err)
 	}
 
-	if serviceRun.IsCompleted() {
-		service.Status.Phase = serviceRun.Status.Phase
-
-		if err := r.Status().Update(ctx, service); err != nil {
-			logger.Error(err, "Could not update Service")
-			return reconcile.Result{}, err
-		}
-		return ctrl.Result{}, nil
-	}
-
-	return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// to make use of .spec.replicaSetUID as a FieldSelector, we need to provide an index for that field
+	// if err := mgr.GetFieldIndexer().IndexField(context.Background(), &v1alpha1.Service{}, ".spec.ownerUID", func(rawObj client.Object) []string {
+	// 	service := rawObj.(*v1alpha1.Service)
+	// 	return []string{string(service.Spec.OwnerUID)}
+	// }); err != nil {
+	// 	return err
+	// }
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Service{}).
 		Complete(r)
 }
 
-func (r *ServiceReconciler) createServiceRun(ctx context.Context, service *v1alpha1.Service) (string, error) {
+func (r *ServiceReconciler) createServiceRun(ctx context.Context, service *v1alpha1.Service) (*v1alpha1.ServiceRun, error) {
 	serviceRun := &v1alpha1.ServiceRun{
 		ObjectMeta: metav1.ObjectMeta{
-			Annotations: map[string]string{
-				"keptn.sh/application": service.Spec.ApplicationName,
-				"keptn.sh/service":     service.Name,
+			Annotations: service.Annotations,
+			Name:        service.GetServiceRunName(),
+			Namespace:   service.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: service.APIVersion,
+					Kind:       service.Kind,
+					Name:       service.Name,
+					UID:        service.UID,
+				},
 			},
-			Name:      service.Name + "-" + r.generateSuffix(),
-			Namespace: service.Namespace,
 		},
 		Spec: v1alpha1.ServiceRunSpec{
-			ServiceName: service.Name,
+			ApplicationName:    service.Spec.ApplicationName,
```

Comment:
let's also add the reference to the replicaSet/Pod (i.e. the field currently called `Owner` in the `Service` spec), so the controller can keep track of the status of the referenced object after the pre deployment tasks have been finished